src/pages/profiles/ProfileSelector.tsx

```diff
@@ -70,109 +36,99 @@ const ProfileSelector: FC<Props> = ({ project, selected, setSelected }) => {
-                    >
-                      Apply
-                    </Button>
-                  </div>
-                </Fragment>
-              ))}
-            />
-          </div>
-        </Col>
-        <Col size={6} className="p-profile-select-box">
-          <i className="p-profile-select-title">Applied profiles</i>
-          <hr />
-          <div className="p-profile-select-list">
-            <List
-              items={selected.map((name, i) => {
-                const profile = profiles.find(
-                  (profile) => profile.name === name
-                );
-                if (!profile) {
-                  return null;
+      {selected.map((value, index) => (
+        <Row key={value}>
+          <Col size={8}>
+            <Select
+              id={`profile-${index}`}
+              help={
+                index > 0 &&
+                index === selected.length - 1 &&
+                "Additional profiles override settings from profiles above"
+              }
+              label={`Profile ${index + 1}`}
+              onChange={(e) => {
+                selected[index] = e.target.value;
+                setSelected(selected);
+              }}
+              options={profiles
+                .filter(
+                  (profile) =>
+                    !selected.includes(profile.name) ||
+                    selected.indexOf(profile.name) === index
+                )
+                .map((profile) => {
+                  return {
+                    label: profile.name,
+                    value: profile.name,
+                  };
+                })}
+              value={value}
+            ></Select>
+          </Col>
+          {index > 0 && (
+            <Col size={4}>
+              <Button
+                appearance="link"
+                className="profile-action-btn"
+                onClick={() => {
+                  const newSelection = [...selected];
+                  newSelection.splice(index, 1);
+                  newSelection.splice(index - 1, 0, value);
+                  setSelected(newSelection);
+                }}
+                type="button"
+                aria-label="move profile up"
+                title="move profile up"
+              >
+                <Icon name="chevron-up" />
+              </Button>
+              <Button
+                appearance="link"
+                className="profile-action-btn"
+                onClick={() => {
+                  const newSelection = [...selected];
+                  newSelection.splice(index, 1);
+                  newSelection.splice(index + 1, 0, value);
+                  setSelected(newSelection);
+                }}
+                type="button"
+                aria-label="move profile down"
+                title="move profile down"
+                disabled={index === selected.length - 1}
+              >
+                <Icon name="chevron-down" />
+              </Button>
```

Comment:
Is there a particular reason to disallow the reordering of the first profile? It looks odd and unbalanced without the arrows on the first profile as well. I can understand that it cannot be deleted, but it should still be possible to move it down, right? Maybe you can hide the arrows only if there is just 1 profile.

Reply:
True, that makes sense to also show them on the 1st profile in case there are multiple.
Just pushed all the changes @lorumic can you give it another pass?